### ISSUE TITLE
vendored version of six.py in pypi wheel breaks newer versions of pip

### ISSUE THREAD

**Issue by rhelmot** [pysmt/pysmt#652]
The wheel that gets installed when you say `pip install pysmt` replaces the global version of six:

```
$ python -c 'import six; print(six, six.__version__)'
<module 'six' from '/home/audrey/.virtualenvs/angr3/lib/python3.8/site-packages/six.py'> 1.14.0                                                                                                            
$ pip install pysmt
Collecting pysmt                                                                                                                                                                                           
  Using cached PySMT-0.9.0-py2.py3-none-any.whl (317 kB)                                                                                                                                                   
Requirement already satisfied: six in /home/audrey/.virtualenvs/angr3/lib/python3.8/site-packages (from pysmt) (1.14.0)                                                                                    
Installing collected packages: pysmt                                                                 
Successfully installed pysmt-0.9.0                                                                   
$ python -c 'import six; print(six, six.__version__)'
<module 'six' from '/home/audrey/.virtualenvs/angr3/lib/python3.8/site-packages/six.py'> 1.10.0 
```

This breaks newer versions of pypi, which depend on six>=1.11:

```
$ pip install sqlalchemy
Traceback (most recent call last):
  File "/home/audrey/.virtualenvs/angr3/bin/pip", line 8, in <module>
    sys.exit(main())
  File "/home/audrey/.virtualenvs/angr3/lib/python3.8/site-packages/pip/_internal/cli/main.py", line 73, in main
    command = create_command(cmd_name, isolated=("--isolated" in cmd_args))
  File "/home/audrey/.virtualenvs/angr3/lib/python3.8/site-packages/pip/_internal/commands/__init__.py", line 96, in create_command
    module = importlib.import_module(module_path)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/audrey/.virtualenvs/angr3/lib/python3.8/site-packages/pip/_internal/commands/install.py", line 24, in <module>
    from pip._internal.cli.req_command import RequirementCommand
  File "/home/audrey/.virtualenvs/angr3/lib/python3.8/site-packages/pip/_internal/cli/req_command.py", line 20, in <module>
    from pip._internal.operations.prepare import RequirementPreparer
  File "/home/audrey/.virtualenvs/angr3/lib/python3.8/site-packages/pip/_internal/operations/prepare.py", line 16, in <module>
    from pip._internal.distributions import (
  File "/home/audrey/.virtualenvs/angr3/lib/python3.8/site-packages/pip/_internal/distributions/__init__.py", line 2, in <module>
    from pip._internal.distributions.wheel import WheelDistribution
  File "/home/audrey/.virtualenvs/angr3/lib/python3.8/site-packages/pip/_internal/distributions/wheel.py", line 5, in <module>
    from pip._internal.utils.wheel import pkg_resources_distribution_for_wheel
  File "/home/audrey/.virtualenvs/angr3/lib/python3.8/site-packages/pip/_internal/utils/wheel.py", line 12, in <module>
    from pip._vendor.six import PY2, ensure_str
ImportError: cannot import name 'ensure_str' from 'six' (/home/audrey/.virtualenvs/angr3/lib/python3.8/site-packages/six.py)
```

(it looks like there is an additional bug in that _pip's_ "vendored" import of six is in fact importing the global version, but that's a problem for another day)

**Comment by marcogario** [comment]
Thanks for reporting this. `six` is not require anymore, as we do not support python 2 anymore, so we might just remove this from the package. I do not have a timeline unfortunately.

**Comment by Scherlac** [comment]
Same here:

```console
root@c5b1d8fcfe68:/opt/orchestrator# head -n 40 /opt/pipvenv/lib/python3.8/site-packages/six.py |grep "__version"
__version__ = "1.16.0"
root@c5b1d8fcfe68:/opt/orchestrator# pip install pysmt==0.9.0
Collecting pysmt==0.9.0
  Downloading PySMT-0.9.0-py2.py3-none-any.whl (317 kB)
     |████████████████████████████████| 317 kB 11.0 MB/s 
Requirement already satisfied: six in /opt/pipvenv/lib/python3.8/site-packages (from pysmt==0.9.0) (1.16.0)
Installing collected packages: pysmt
Successfully installed pysmt-0.9.0
root@c5b1d8fcfe68:/opt/orchestrator# head -n 40 /opt/pipvenv/lib/python3.8/site-packages/six.py |grep "__version"
__version__ = "1.10.0"
```

**Comment by marcogario** [comment]
You should be able to work around the issue by using a pre-release version of pysmt. Any version after this PR should work.
The full list of pre-releases is visible at https://pypi.org/project/PySMT/#history or you can use `pip --pre` to get the latest.

An actual release is long overdue :disappointed:

**Comment by Scherlac** [comment]
Symptom for me was:

``` console
from azure.mgmt.datafactory import DataFactoryManagementClient
...
  File "/opt/pipvenv/lib/python3.7/site-packages/azure/mgmt/datafactory/models/_data_factory_management_client_enums.py", line 29, in <module>
    class AvroCompressionCodec(with_metaclass(_CaseInsensitiveEnumMeta, str, Enum)):
  File "/usr/local/lib/python3.7/enum.py", line 135, in __prepare__
    member_type, first_enum = metacls._get_mixins_(bases)
  File "/usr/local/lib/python3.7/enum.py", line 472, in _get_mixins_
    raise TypeError("new enumerations should be created as "
TypeError: new enumerations should be created as `EnumName([mixin_type, ...] [data_type,] enum_type)`

```

**Comment by marcogario** [comment]
I published a stable version 0.9.5 that includes this fix.
